src/main/kotlin/org/rust/ide/refactoring/changeSignature/RsChangeSignatureConfig.kt

```diff
@@ -90,14 +90,13 @@ class RsChangeFunctionSignatureConfig private constructor(
     val returnType: Ty
         get() = returnTypeDisplay?.type ?: TyUnit
 
-    private val parametersText: String
-        get() {
-            val selfText = listOfNotNull(function.selfParameter?.text)
-            val parametersText = parameters.map { "${it.pat.text}: ${it.typeReference.text}" }
-            return (selfText + parametersText).joinToString(", ")
-        }
+    private fun parametersText(parameters: List<Parameter>): String {
+        val selfText = listOfNotNull(function.selfParameter?.text)
+        val parametersText = parameters.map { "${it.pat.text}: ${it.typeReference.text}" }
+        return (selfText + parametersText).joinToString(", ")
+    }
 
-    fun signature(): String = buildString {
+    fun signature(parameters: List<Parameter>): String = buildString {
```

Comment:
This is.. not pretty, but the problem is that `calculateSignature()` seems to be called before `updateState` for some reason, so this seemed like the easiest solution.

Reply:
Nevermind, I solved it in a different way.